util/billing.go

```diff
@@ -0,0 +1,152 @@
+package util
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/olivere/elastic/v7"
+)
+
+var TimeValidity int64
+
+type ArcUsage struct {
+	ArcID          string `json:"arc_id"`
+	Timestamp      int64  `json:"timestamp"`
+	SubscriptionID string `json:"subscription_id"`
+	Quantity       int    `json:"quantity"`
+	Email          string `json:"email"`
+}
+
+type ArcUsageResponse struct {
+	Accepted      bool   `json:"accepted"`
+	FailureReason string `json:"failure_reason"`
+	ErrorMsg      string `json:"error_msg"`
+	WarningMsg    string `json:"warning_msg"`
+	StatusCode    int    `json:"status_code"`
+	TimeValidity  int64  `json:"time_validity"`
+}
+
+const (
+	envEsURL       = "ES_CLUSTER_URL"
+	arcIdentifier  = "ARC_ID"
+	emailID        = "EMAIL"
+	subscriptionID = "SUBSCRIPTION_ID"
+)
+
+// Middleware function, which will be called for each request
+func BillingMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		// Valid plan
+		if TimeValidity > 0 {
```

Comment:
@bietkul Technically, if TimeValidity > 0, the user is still within the limits of the plan. We can't send a warning till the TimeValidity <= 0.

Reply:
Check the second condition, we're only showing the warning message when left time is less than 24 hrs.